### PR TITLE
Add minimumCompressionSize trait and OpenAPI mapper

### PR DESCRIPTION
**Pull request by mee-aa**
Add the aws.apigateway#minimumCompressionSize integer trait with @range(min: 0, max: 10485760) validation. The OpenAPI mapper converts it to the x-amazon-apigateway-minimum-compression-size extension for REST APIs.

#### Background
* Adds a new `minimumCompressionSize` trait to `smithy-aws-apigateway-traits` that lets customers configure the minimum payload size (in bytes) for API Gateway compression directly in their Smithy model.
* Adds the corresponding OpenAPI mapper in `smithy-aws-apigateway-openapi` that converts the trait to the `x-amazon-apigateway-minimum-compression-size` extension during Smithy-to-OpenAPI conversion.
* Without this trait, customers had to use `jsonAdd` workarounds in `smithy-build.json` to set this value. This is the first of several new API Gateway REST API traits being added.

#### Testing
* Unit tests for the trait class: factory registration, round-trip serialization, boundary values (0 and 10485760)
* Integration test for the OpenAPI mapper: loads a Smithy model with the trait applied, converts to OpenAPI, and verifies the extension is present with the correct value
* All existing tests pass

#### Links
* [API Gateway minimum compression size docs](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-openapi-minimum-compression-size.html)
* Follows the `HttpErrorTrait` pattern for integer traits in the Smithy codebase

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: smithy-aws-apigateway-traits/src/main/resources/META-INF/smithy/aws.apigateway.smithy [hunk 1]
```diff
@@ -354,3 +354,10 @@ enum PayloadFormatVersion {
     /// Specifies 2.0 version of the format used by the authorizer
     V2_0 = "2.0"
 }
+
+
+/// Defines the minimum payload size for compression on the API.
+/// Value must be between 0 and 10485760 (10MB).
+@trait(selector: "service")
```
Comment (kstich) [inline smithy-aws-apigateway-traits/src/main/resources/META-INF/smithy/aws.apigateway.smithy]: This should have:

```smithy
@internal
@tags(["internal"])
```

as it's an implementation detail of the service.

It should also be placed in the file up with the other traits towards the top of the file.

File: smithy-aws-apigateway-openapi/src/test/resources/software/amazon/smithy/aws/apigateway/openapi/minimum-compression-size.json
Comment (kstich) [inline smithy-aws-apigateway-openapi/src/test/resources/software/amazon/smithy/aws/apigateway/openapi/minimum-compression-size.json:1]: Replace with a .smithy file

```smithy
$version: "2.0"

namespace smithy.example

use aws.apigateway#minimumCompressionSize
use aws.protocols#restJson1

@restJson1
@minimumCompressionSize(10240)
service Service {
    version: "2006-03-01"
    operations: [
        Operation1
    ]
}

@http(uri: "/1", method: "GET")
operation Operation1 {}
```

File: smithy-aws-apigateway-openapi/src/test/java/software/amazon/smithy/aws/apigateway/openapi/AddMinimumCompressionSizeTest.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import software.amazon.smithy.openapi.OpenApiConfig;
+import software.amazon.smithy.openapi.fromsmithy.OpenApiConverter;
+import software.amazon.smithy.openapi.model.OpenApi;
+
+public class AddMinimumCompressionSizeTest {
+    @Test
+    public void addsMinimumCompressionSize() {
+        Model model = Model.assembler()
+                .discoverModels(getClass().getClassLoader())
+                .addImport(getClass().getResource("minimum-compression-size.json"))
```
Comment (kstich) [inline smithy-aws-apigateway-openapi/src/test/java/software/amazon/smithy/aws/apigateway/openapi/AddMinimumCompressionSizeTest.java]:
```suggestion
                .addImport(getClass().getResource("minimum-compression-size.smithy"))
```

**Comment by sugmanue** [comment]
The change looks good to me, it's just missing documentation here https://github.com/smithy-lang/smithy/blob/main/docs/source-2.0/aws/amazon-apigateway.rst

File: smithy-aws-apigateway-traits/src/main/java/software/amazon/smithy/aws/apigateway/traits/MinimumCompressionSizeTrait.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import software.amazon.smithy.model.FromSourceLocation;
+import software.amazon.smithy.model.SourceLocation;
+import software.amazon.smithy.model.node.Node;
+import software.amazon.smithy.model.node.NumberNode;
+import software.amazon.smithy.model.shapes.ShapeId;
+import software.amazon.smithy.model.traits.AbstractTrait;
+import software.amazon.smithy.model.traits.Trait;
+
+/**
+ * Defines the minimum payload size for compression on an API Gateway REST API.
```
Comment (sugmanue) [inline smithy-aws-apigateway-traits/src/main/java/software/amazon/smithy/aws/apigateway/traits/MinimumCompressionSizeTrait.java]:
```suggestion
 * Defines the minimum payload size in bytes at which compression is applied on an API Gateway REST API.
```

File: smithy-aws-apigateway-traits/src/main/resources/META-INF/smithy/aws.apigateway.smithy [hunk 4]
```diff
@@ -354,3 +354,10 @@ enum PayloadFormatVersion {
     /// Specifies 2.0 version of the format used by the authorizer
     V2_0 = "2.0"
 }
+
```
Comment (sugmanue) [inline smithy-aws-apigateway-traits/src/main/resources/META-INF/smithy/aws.apigateway.smithy]: Nit, remove extra newline.
```suggestion
```

**Review by kstich — CHANGES_REQUESTED** [review]
This also needs spec updates:

* [Document the trait in the API GW traits](https://github.com/smithy-lang/smithy/blob/367c4a5ac4f9bae217c0f00b45344f0872d7491d/docs/source-2.0/aws/amazon-apigateway.rst?plain=1)
* [The Converting Smithy to OpenAPI guide needs an update](https://github.com/smithy-lang/smithy/blob/fd4987db3e7563368cf589ea5892407d03bb2bf2/docs/source-2.0/guides/model-translations/converting-to-openapi.rst?plain=1#L1627).

File: smithy-aws-apigateway-traits/src/main/resources/META-INF/smithy/aws.apigateway.smithy [hunk 5]
```diff
@@ -354,3 +354,10 @@ enum PayloadFormatVersion {
...
+
+/// Defines the minimum payload size for compression on the API.
+/// Value must be between 0 and 10485760 (10MB).
```
Comment (sugmanue) [inline smithy-aws-apigateway-traits/src/main/resources/META-INF/smithy/aws.apigateway.smithy]:
```suggestion
/// Defines the minimum payload size in bytes at which compression is applied on an API Gateway REST API.
/// Value must be between 0 and 10485760 (10MB).
```

**Review by kstich — APPROVED** [review]

File: docs/source-2.0/aws/amazon-apigateway.rst [hunk 6]
```diff
@@ -240,6 +240,46 @@ Value type
     customers.
 
 
+.. smithy-trait:: aws.apigateway#minimumCompressionSize
+.. _aws.apigateway#minimumCompressionSize-trait:
+
+-------------------------------------------------
+``aws.apigateway#minimumCompressionSize`` trait
+-------------------------------------------------
```
Comment (kstich) [inline docs/source-2.0/aws/amazon-apigateway.rst]:
```suggestion
-----------------------------------------------
``aws.apigateway#minimumCompressionSize`` trait
-----------------------------------------------
```

File: smithy-aws-apigateway-traits/src/test/java/software/amazon/smithy/aws/apigateway/traits/MinimumCompressionSizeTraitTest.java [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+    @Test
+    public void roundTripsFromNode() {
+        MinimumCompressionSizeTrait trait = new MinimumCompressionSizeTrait(0);
+
+        assertThat(trait.getValue(), equalTo(0));
+        assertThat(trait.toNode(), equalTo(Node.from(0)));
+        assertThat(new MinimumCompressionSizeTrait.Provider()
+                .createTrait(ShapeId.from("ns.foo#Bar"), trait.toNode()),
+                equalTo(trait));
+    }
```
Comment (kstich) [inline smithy-aws-apigateway-traits/src/test/java/software/amazon/smithy/aws/apigateway/traits/MinimumCompressionSizeTraitTest.java]:
```suggestion
```

**Review by kstich — CHANGES_REQUESTED** [review]

**Comment by kstich** [comment]
Looks like there's a broken cross reference in the docs.

File: .changes/next-release/feature-d2ab780e96f07b4a861e1e3dddb489c539f6f642.json [hunk 8]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "feature",
+  "description": "Add aws.apigateway#minimumCompressionSize trait and OpenAPI mapper",
```
Comment (kstich) [inline .changes/next-release/feature-d2ab780e96f07b4a861e1e3dddb489c539f6f642.json]:
```suggestion
  "description": "Added `aws.apigateway#minimumCompressionSize` trait and OpenAPI mapper",
```

File: docs/source-2.0/aws/amazon-apigateway.rst [hunk 9]
```diff
@@ -240,6 +240,43 @@ Value type
     customers.
 
 
+.. smithy-trait:: aws.apigateway#minimumCompressionSize
+.. _aws.apigateway#minimumCompressionSize-trait:
+
+-------------------------------------------------
+``aws.apigateway#minimumCompressionSize`` trait
```
Comment (kstich) [inline docs/source-2.0/aws/amazon-apigateway.rst]:
```suggestion
```

**Review by kstich — APPROVED** [review]
